gnat2goto/driver/tree_walk.adb

```diff
@@ -4144,53 +4050,109 @@ package body Tree_Walk is
    -- Do_Slice --
    --------------
 
+   --  The following build an expression representing slice
+   --  orig(start .. end)
+   --  Let ArrT := struct { int first; int last; T* data; }
+   ----------------------------------------------------------------------------
+   --  ArrT slice_expr(ArrT orig) {
+   --    T* new_data = data + (start - orig.first);
+   --    ArrT temp_array = {.first=start, .last=end, .data=new_data};
+   --    return temp_array;
```

Comment:
Repeating the question about null slices here (not sure where the right place to handle those is actually, but this looks like one place to look at?)

Reply:
As above.